site/app/libraries/socket/Server.php

```diff
@@ -72,21 +77,19 @@ private function checkAuth(ConnectionInterface $conn): bool {
 
     /**
      * Push a given message to all or all-but-sender connections
+     * @param ConnectionInterface $from
+     * @param string $content
+     * @param string $course_name
+     * @return void
      */
-    private function broadcast(ConnectionInterface $from, string $content, string $course_name): void {
+    private function broadcast(ConnectionInterface $from, $content, $course_name): void {
```

Comment:
why remove the inline type hints?

Reply:
i added those type hints in the annotation right above the function, thought that was enough then.

Reply:
It's not, and we should be having inline types as much as possible, especially new classes / files, where we can do `declare(strict_types=1);` at the top, like here. If a file is declared strict, calling a function with the wrong argument types will cause an error to be raised if you call the function with the wrong types (and if you have the inline types but not strict typing, PHP will try and silently coerce types which I dislike).

See [this doc](https://www.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.strict) for more info on strict typing.

Reply:
will add inline type declaration.